pkg/provider/azure_loadbalancer.go

```diff
@@ -1887,11 +1891,11 @@ func (az *Cloud) reconcileBackendPools(clusterName string, service *v1.Service,
 				for _, ipConf := range *bp.BackendIPConfigurations {
 					ipConfID := to.String(ipConf.ID)
 					nodeName, _, err := az.VMSet.GetNodeNameByIPConfigurationID(ipConfID)
-					if err != nil {
+					if err != nil && !errors.Is(err, cloudprovider.InstanceNotFound) {
 						return false, false, err
 					}
 					if nodeName == "" {
-						// VM may under deletion
+						// VM may be under deletion or have already been deleted
```

Comment:
Similiar concern: if the VM has already been deleted, why do we still need the corresponding IP config to be in the backend pool?

Reply:
good catch, we should continue to next step on such cases